azkaban-common/src/main/java/azkaban/executor/container/KubernetesContainerizedImpl.java

```diff
@@ -788,28 +922,34 @@ String getFlowContainerDiskRequest(final Map<String, String> flowParam) {
    * requested resource):
    * 1) if resource 1 >= resource 2, return 1;
    * 2) if resource 1 < resource 2, return -1;
-   * 3) if resource 1 cannot be compared with resource 2, e.g. parse error, mis-matching resource
-   * types, return 0;
+   * 3) if resource 1 cannot be compared with resource 2, e.g. parse error, return 0;
    * The format of a Kubernetes quantity indicates the type of resource, e.g. cpu, memory.
    * @param resource1
    * @param resource2
    * @return
    */
-  private int compareResources (final String resource1, final String resource2) {
+  private static int compareResources(final String resource1, final String resource2) {
     try {
       final Quantity quantity1 = new Quantity(resource1), quantity2 = new Quantity(resource2);
-      if (quantity1.getFormat() == quantity2.getFormat()) {
-        return quantity1.getNumber().compareTo(quantity2.getNumber()) < 0 ? -1 : 1;
-      }
-    } catch (final QuantityFormatException qe) { // only user requested resource in flow
+      return quantity1.getNumber().compareTo(quantity2.getNumber()) < 0 ? -1 : 1;
+    } catch (final Exception e) { // only user requested resource in flow
```

Comment:
Why replace "QuantityFormatException" with the general one?

Reply:
it's not recommended to catch general exception, as it would ignore InterruptException and the webserver would then can not be gracefully shutdown. (I know we use those catches everywhere but it's a good habit to have to avoid such, we should start to explicitly state exact exception.)